providers/os/resources/parse.go

```diff
@@ -130,6 +132,190 @@ func (s *mqlParseJson) params(content string) (interface{}, error) {
 	return res, nil
 }
 
+func initParseXml(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if err := fileFromPathOrContent(runtime, args); err != nil {
+		return nil, nil, err
+	}
+
+	return args, nil, nil
+}
+
+func (s *mqlParseXml) id() (string, error) {
+	if s.File.Data == nil {
+		return "", errors.New("no file provided for parse.json")
+	}
+
+	file := s.File.Data
+	return file.Path.Data, nil
+}
+
+func (s *mqlParseXml) content(file *mqlFile) (string, error) {
+	c := file.GetContent()
+	return c.Data, c.Error
+}
+
+type xmlElem struct {
+	attributes map[string]string
+	children   []*xmlElem
+	data       string
+	isElement  bool
+}
+
+func (x *xmlElem) addAttr(a []xml.Attr) {
+	if len(a) == 0 {
+		return
+	}
+	x.attributes = map[string]string{}
+	for ai := range a {
+		attr := a[ai]
+		x.attributes[attrName(attr.Name)] = attr.Value
+	}
+}
+
+func (x *xmlElem) UnmarshalXML(decoder *xml.Decoder, start xml.StartElement) error {
+	x.data = start.Name.Local
+	x.isElement = true
+	path := []*xmlElem{
+		x,
+	}
+	path[0].addAttr(start.Attr)
+
+	for {
+		token, err := decoder.Token()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+
+		switch elem := token.(type) {
+		case xml.StartElement:
+			nu := &xmlElem{
+				data:      elem.Name.Local,
+				isElement: true,
+			}
+			nu.addAttr(elem.Attr)
+			parent := path[len(path)-1]
+			parent.children = append(parent.children, nu)
+			path = append(path, nu)
+		case xml.EndElement:
+			path = path[:len(path)-1]
+		case xml.CharData:
+			cur := path[len(path)-1]
+			v := strings.TrimSpace(string(elem))
+			if v != "" {
+				cur.children = append(cur.children, &xmlElem{
+					data:      v,
+					isElement: false,
+				})
+			}
+		}
+	}
+}
+
+func (x *xmlElem) _params() (string, bool, map[string]any) {
+	if !x.isElement {
+		return x.data, x.isElement, nil
+	}
+	res := map[string]any{}
+	for k, v := range x.attributes {
+		res["@"+k] = v
+	}
+
+	for i := range x.children {
+		child := x.children[i]
+		data, isElem, params := child._params()
+
+		// text data is added flat
```

Comment:
nit I wonder if we should separate the text content into a slice to simplify this code, maybe adding the slice to the `xmlElem` and append it in `UnmarshalXML()` for the `xml.CharData` case.